### DISCUSSION TITLE
Using zigpy to implement a device

### DISCUSSION THREAD

**Discussion by PaulKierstead** [zigpy/zigpy#732] [Q&A]
My deepest apologies if this has been asked and answered, but I don't see it. I don't want to *control* Zigbee devices, I want to make a custom one emulating a standard class of device. I'm pretty familiar with Xbee Zigbee devices using custom protocols, but not the HA stack. Has anyone used zigpy to fill in the stack and have an example I can start with?

**Comment by Adminiuga** [comment]
This was not done, so there aren't any examples. Technically, the framework is there, and you just need to implement handlers for general commands (like some other entity reading or writing an attribute of the cluster) and cluster specific commands, like receive ing an on command on inn off cluster. 
There other part missing, is joining the network as a device. Currently zigpy and radio libs implement coordinator mode only. Although, adding a support jobs n as a device, shouldn't be that hard. 

Start reading the ZCL specs to understand ZCL app level, commands etc. Then zdo
